### PR TITLE
Remove Qt code below 5.15

### LINKED ISSUE #35072: Remove pre-Qt5.15 related code
Examine the Qt directory and look for any code that checks for the Qt version. You can search for the `QT_VERSION` across the qt directory. 

 - Remove those that are for compatability with Qt < 5.15
 - Remove our custom SignalBlocker: https://github.com/mantidproject/mantid/blob/main/qt/widgets/common/inc/MantidQtWidgets/Common/SignalBlocker.h
 - Set the minimum Qt version in CMake to 5.15

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR removes all references to Qt versions which are less than the current version of Qt 5.15 .

**To test:**
Code review
Open the Muon interfaces and make sure the dock widgets are in the expected location and have the expected size. The same needs to be done with the FigureManager.
Check external plotting still works on the Muon interfaces

Fixes #35072

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/CMakeLists.txt [hunk 1]
```diff
@@ -20,19 +20,19 @@ set(QTPROPERTYBROWSER_SRCS
 # moc'd files will end up in build directory, so add to include path
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-# Need to use a string of calls to the qt4_generate_moc macro instead of qt4_wrap_cpp because the latter produces files
+# Need to use a string of calls to the qt5_generate_moc macro instead of qt5_wrap_cpp because the latter produces files
 # with a cxx suffix when QtPropertyBrowser requires cpp
 
-qt4_generate_moc(src/qtpropertybrowser.h ${CMAKE_CURRENT_BINARY_DIR}/moc_qtpropertybrowser.cpp)
```
Comment (martyngigg) [inline qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/CMakeLists.txt:26]: Do we know how this worked until now? We removed using Qt4 2 years ago...

**Review by martyngigg — CHANGES_REQUESTED** [review]
Great to clean up a lot of old code. Some comments inline...

File: qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/CMakeLists.txt (hunk 1 above)
Comment (robertapplin) [reply qt/widgets/common/inc/MantidQtWidgets/Common/QtPropertyBrowser/CMakeLists.txt:26]: The `qt4_generate_moc` command seems to generate the moc files as expected (same as the `qt5_generate_moc` command).

The conversation on [here](https://development.qt-project.narkive.com/odygyYSZ/installing-qt5config-cmake-from-the-qt-repo), and the code in this link seems to indicate that the `qt4_generate_moc` macro raps around the `qt5_generate_moc` macro

https://cmake.org/pipermail/cmake/2012-December/052970.html

**Comment by robertapplin** [comment]
I've removed as much "MantidPlot" references as I dare, so this is ready for another look over @martyngigg . I think most if not all Qt4 code should now be gone

File: docs/source/plotting/index.rst [hunk 2]
```diff
@@ -476,7 +476,7 @@ Custom Colormap (MantidWorkbench)
 You can view the premade Colormaps `here <https://matplotlib.org/2.2.3/gallery/color/colormap_reference.html?highlight=colormap>`_.
 These Colormaps can be registered and remain for the current session, but need to be rerun if Mantid has been reopened. Choose the location to Save your Colormap file wisely, outside of your MantidInstall folder!
 
-The following methods show how to Load, Convert from MantidPlot format, Create from Scratch and Visualise a Custom Colormap.
```
Comment (martyngigg) [inline docs/source/plotting/index.rst:479]: I think this actually do need to say MantidPlot. @DanielMurphy22 Am I right in saying these were about converting from the old custom format to matplotlib?

**Review by martyngigg — COMMENTED** [review]
I haven't got completed through the remaining files yet but just a couple more comments.

File: docs/source/plotting/index.rst (hunk 2 above)
Comment (DanielMurphy22) [reply docs/source/plotting/index.rst]: Yes indeed this is for converting from a MantidPlot colormap to one that can be loaded for Workbench

**Review by martyngigg — COMMENTED** [review]
Just one final question...

**Review by martyngigg — APPROVED** [review]
Looks great! It's nice to be rid of a lot of dead code.

**Pull request merged**